provisioning/miniprov/hare_mp/main.py

```diff
@@ -243,11 +249,47 @@ def is_cluster_running() -> bool:
     return os.system('hctl status >/dev/null') == 0
 
 
+def nr_services() -> int:
+    cmd = ['hctl', 'status', '--json']
+    cluster_info = json.loads(execute(cmd))
+    # Don't include hax, count it just once later.
+    services = ['confd', 'ioservice', 's3server']
+    svcs_nr = 0
+    for node in cluster_info['nodes']:
+        for svc in node['svcs']:
+            if svc['name'] in services:
+                svcs_nr += 1
+    return (svcs_nr + 1)
+
+
+def all_services_started(url: str, nr_svcs: int) -> bool:
+    conf = ConfStoreProvider(url)
+    hostname = conf.get_hostname()
+    res = requests.get((f'http://127.0.0.1:8500/v1/kv/{hostname}/'
+                        f'processes?recurse=True'))
+    status_data = json.loads(res.text)
+    statuses = []
+    for val in status_data:
+        state = base64.decodestring(bytes(val['Value'], 'utf-8'))
+        statuses.append(json.loads(state.decode('utf8'))['state'])
+    started = [status == 'M0_CONF_HA_PROCESS_STARTED' for status in statuses]
+    if len(started) != nr_svcs:
+        return False
+    return all(started)
+
+
 def bootstrap_cluster(path_to_cdf: str, domkfs=False):
     if domkfs:
-        return os.system('hctl bootstrap --mkfs ' + path_to_cdf)
+        rc = os.system('hctl bootstrap --mkfs ' + path_to_cdf)
     else:
-        return os.system('hctl bootstrap ' + path_to_cdf)
+        rc = os.system('hctl bootstrap ' + path_to_cdf)
+    return rc
+
+
+def wait_for_cluster_start(url: str):
+    nr_svcs = nr_services()
```

Comment:
I would vote for logging here. The script will hang by some reason and the end user will have to guess what's happening.

Reply:
Okay, makes sense.